### PR TITLE
feat: complete ruby support

### PR DESCRIPTION
**Pull request by urbit-pilled**
/claim #56 

This PR adds more unit tests, and should provide more complete Ruby language support.

## Summary by CodeRabbit

- **New Features**
  - Enhanced Ruby language support with updated class, module, and method handling in test functions.
  - Improved Ruby syntax parsing with new tokens and rules for better code interpretation.
- **Bug Fixes**
  - Adjusted Ruby language test cases to align with updated language patterns.
- **Refactor**
  - Updated internal grammar and node types for Ruby to enhance clarity and functionality.
- **Documentation**
  - Extended documentation for new test functions and grammar rules specific to Ruby language enhancements.
- **Style**
  - Implemented Google style guide in ESLint configuration for consistent code styling.

### CODE REVIEW

**Comment by morgante** [comment]
@urbit-pilled How's this looking? Ready for review?

**Comment by urbit-pilled** [comment]
should finally be ready for review now

File: crates/language/src/ruby.rs [hunk 1]
```diff
@@ -59,7 +59,10 @@ impl Language for Ruby {
     }
 
     fn snippet_context_strings(&self) -> &[(&'static str, &'static str)] {
-        &[("", "")]
+        &[
+            ("", ""),
+            ("case GRIT_VARIABLE\n", "\nend")
```
Comment (morgante) [inline crates/language/src/ruby.rs:64]: I suspect we need more contexts. Can you verify these cases work?
- Matching on a method by itself (`def $_: $_` inside a class)
- Matching an hash property (`name: "Alice"` inside `person = {name: "Alice", age: 25, city: "New York"}`)

**Review by morgante — APPROVED** [review]
Looks good to merge, but we will need a few more cases covered.

File: crates/language/src/ruby.rs (hunk 1 above)
Comment (urbit-pilled) [reply crates/language/src/ruby.rs:64]: I updated the contexts and tests in https://github.com/getgrit/gritql/pull/286/commits/e3e4bd1436438b0cfaea7a8fc4a21d4dd2f61253#

- matching a method by itself inside a class is added in test `ruby_class_method`
- matching a hash property is added in test `ruby_hash`

**Pull request merged**